docs/decisions/0001-purpose-of-this-repo.rst

```diff
@@ -6,52 +6,42 @@ Status
 *******
 
-TODO: Add context of what led to the creation of this repo.
+With the `deprecation of ecommerce`_, Axim launched a funded project to develop capabilities of connecting Open edX to external ecommerce system. The intention is to connect Open edX with a third-party e-commerce platform, enabling a straightforward implementation process and making the e-commerce service accessible for various initiatives. The request for proposal can be found here: `tCRIL Funded Contribution - WooCommerce Discovery`_.
 
-.. This section describes the forces at play, including technological, political, social, and project local. These forces are probably in tension, and should be called out as such. The language in this section is value-neutral. It is simply describing facts.
+For this integration project, we create a `WooCommerce Discovery document`_, which proposes the creation of a WordPress plugin and an API to generate the necessary actions on the Open edX side. The following image illustrates the expected interactions.
 
-Decision
-********
+.. image:: ../_static/enrollment-diagram.png
+  :alt: Sequence diagram of purchase using Woocommerce Integration.
 
-We will create a repository...
+Sequence diagram of purchase using Woocommerce Integration.
 
-TODO: Clearly state how the context above led to the creation of this repo.
+Decision
+********
 
-.. This section describes our response to these forces. It is stated in full sentences, with active voice. "We will …"
+* We have created this repository, an installable Django plugin in Open edX, to provide several endpoints related to the necessary operations on the platform to offer the e-commerce service.
+* The main objective of this API is to provide endpoints for the fulfillment and refund of enrollments.
+* We want the use of the API to be wider than integration with WooCommerce. The idea is to be useful for other integrations with e-commerce services.
 
 Consequences
 ************
 
-TODO: Add what other things will change as a result of creating this repo.
-
-.. This section describes the resulting context, after applying the decision. All consequences should be listed here, not just the "positive" ones. A particular decision may have positive, negative, and neutral consequences, but all of them affect the team and project in the future.
+* You need to install this plugin to access its functionalities.
 
 Rejected Alternatives
 *********************
 
-TODO: If applicable, list viable alternatives to creating this new repo and give reasons for why they were rejected. If not applicable, remove section.
-
-.. This section lists alternate options considered, described briefly, with pros and cons.
+* Create this API directly in the core of the platform. (However, depending on its usage, this could change).
+* Use the `current enrollment API`_ because user requests only support operations for the current user, and the Server-to-server calls may only deactivate or modify existing enrollments' mode.
```

Comment:
@pshiu I think those specific business logic limitations make that endpoint not a great REST API, after speaking with people in community and at 2U, there is a desire to have an endpoint that is powerful and generic so that business logic can live up-stream in the various client apps.  Parts of 2U for example want to be able to manage the business logic of whether or not someone should have access and level of access upstream in something like salesforce and then be able to use a generic API to sync that data into the open edx system.

I think if we ensure that the permissions for the new APIs are pretty strict a generic endpoint could be a powerful way to enable better integrations with 3rd party business systems.

Reply:
Hey @feanil! Makes sense, we should add what you said to the ADR. What I'm trying to convey:

1. If the current fulfillment endpoint has business logic limitations that are prompting our move to a new endpoint, I think this is a great place to provide examples of those limitations.
2. I read the example limitation in the second rejected alternative of the ADR to be, "the current fulfillment endpoint doesn't allow other Open edX services to add a new enrollment", but from the code I'm not sure why this is true.

Reply:
Makes sense, @MaferMazu can you add some of the context above to the details for rejected alternatives?

Reply:
Thanks to both of you. @pshiu, you are right about we can create a course enrollment with that endpoint, but I agree with @feanil about having a more flexible API. And yes, @feanil, I can give more context about that rejected alternative.